src/textual/_work_decorator.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from functools import partial, wraps
+from typing import TYPE_CHECKING, Callable, Coroutine, TypeVar, Union, cast, overload
+
+from typing_extensions import ParamSpec, TypeAlias
+
+if TYPE_CHECKING:
+    from .worker import Worker
+
+
+FactoryParamSpec = ParamSpec("FactoryParamSpec")
+DecoratorParamSpec = ParamSpec("DecoratorParamSpec")
+ReturnType = TypeVar("ReturnType")
+
+Decorator: TypeAlias = Callable[
+    [
+        Union[
+            Callable[DecoratorParamSpec, ReturnType],
+            Callable[DecoratorParamSpec, Coroutine[None, None, ReturnType]],
+        ]
+    ],
+    Callable[DecoratorParamSpec, "Worker[ReturnType]"],
+]
+
+
+@overload
+def work(
+    method: Callable[FactoryParamSpec, Coroutine[None, None, ReturnType]]
+) -> Callable[FactoryParamSpec, "Worker[ReturnType]"]:
+    ...
+
+
+@overload
+def work(
+    method: Callable[FactoryParamSpec, ReturnType]
+) -> Callable[FactoryParamSpec, "Worker[ReturnType]"]:
+    ...
+
+
+@overload
+def work(*, exclusive: bool = False) -> Decorator[..., ReturnType]:
+    ...
+
+
+def work(
+    method: Callable[FactoryParamSpec, ReturnType]
+    | Callable[FactoryParamSpec, Coroutine[None, None, ReturnType]]
+    | None = None,
+    *,
+    name: str = "",
+    group: str = "default",
+    exclusive: bool = False,
+    exit_on_error: bool = True,
+) -> Callable[FactoryParamSpec, Worker[ReturnType]] | Decorator:
+    """Worker decorator factory."""
+
+    def decorator(
+        method: (
+            Callable[DecoratorParamSpec, ReturnType]
+            | Callable[DecoratorParamSpec, Coroutine[None, None, ReturnType]]
+        )
+    ) -> Callable[DecoratorParamSpec, Worker[ReturnType]]:
+        """The decorator."""
+
+        @wraps(method)
+        def decorated(
+            *args: DecoratorParamSpec.args, **kwargs: DecoratorParamSpec.kwargs
+        ) -> Worker[ReturnType]:
+            """The replaced callable."""
+            from .dom import DOMNode
+
+            self = args[0]
+            assert isinstance(self, DOMNode)
+
+            positional_arguments = ", ".join(repr(arg) for arg in args[1:])
+            keyword_arguments = ", ".join(
+                f"{name}={value!r}" for name, value in kwargs.items()
+            )
+            tokens = [positional_arguments, keyword_arguments]
+            worker_description = (
+                f"{method.__name__}({', '.join(token for token in tokens if token)})"
```

Comment:
Faster:

```suggestion
                f"{method.__name__}({', '.join([token for token in tokens if token])})"
```

_Probably_ even faster:

```suggestion
                f"{method.__name__}({', '.join(filter(None, tokens))})"
```

Reply:
True, but not necessarily clearer.